vehicle/mkvi/software/throttle/throttle.c

```diff
@@ -0,0 +1,322 @@
+    .git_sha = STABLE_GIT_COMMIT,
+};
+
+/*
+ * Sets the torque request in the motor controller command message
+ */
+#define SET_TORQUE_REQUEST(torque) \
+    (m192_command_message.torque_command = (torque))
+
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+#define MAX(a, b) (((a) > (b)) ? (a) : (b))
+
+// all possible throttle states
+enum State {
+    THROTTLE_IDLE,
+    THROTTLE_RUN,
+    THROTTLE_OUT_OF_RANGE,
+    THROTTLE_POSITION_IMPLAUSIBILITY,
+    THROTTLE_BRAKE_PRESSED
+};
+
+struct throttle_state_s {
+    volatile bool send_can;
+    uint16_t implausibility_fault_counter;
+    bool brake_pressed;
+} throttle_state = { 0 };
+
+void timer0_isr(void) {
+    throttle_state.send_can = true;
+}
+
+volatile bool run_1ms = false;
+
+void timer1_isr(void) {
+    run_1ms = true;
+}
+
+void pcint0_callback(void) {
+    throttle.ss_is = !gpio_get_pin(SS_IS);
+}
+
+static int16_t get_throttle_travel(const throttle_potentiometer_s* throttle) {
+    /*
+        Read value from throttle potentiometer, maps it to value between 0 and
+        255 where 255 = 100% throttle Returns an int16_t representing pedal
+        travel
+    */
+    int16_t throttle_raw = adc_read(throttle->adc_pin);
+    throttle_raw >>= 2; // TODO: ask why this is here
+    int16_t range = throttle->throttle_max - throttle->throttle_min;
+    float position_pct
+        = (float)(throttle_raw - throttle->throttle_min) / (float)range;
+    return floor(position_pct * 255);
+}
+
+static bool check_out_of_range(int16_t* pos_1, int16_t* pos_2) {
+    /*
+        Check if sensors return a value out of range (<0 or >255)
+        Args:
+            pos_1: int16_t sensor value of throttle pot 1
+            pos_2: int16_t sensor value of throttle pot 2
+        Returns:
+            implausibility: boolean if throttle out of range or not
+    */
+    bool implausibility = false;
+
+    // Check 1st pot
+    if (*pos_1 > 255) {
+        *pos_1 = 255;
+        throttle.throttle_state = THROTTLE_OUT_OF_RANGE;
+        throttle_debug.throttle_out_of_range = true;
+        implausibility = true;
+    } else if (*pos_1 < 0) {
+        *pos_1 = 0;
+        throttle.throttle_state = THROTTLE_OUT_OF_RANGE;
+        throttle_debug.throttle_out_of_range = true;
+        implausibility = true;
+    } else {
+        throttle_debug.throttle_out_of_range = false;
+    }
+
+    // Check 2nd pot
+    if (*pos_2 > 255) {
+        *pos_2 = 255;
+        throttle.throttle_state = THROTTLE_OUT_OF_RANGE;
+        throttle_debug.throttle_out_of_range = true;
+        implausibility = true;
+    } else if (*pos_2 < 0) {
+        *pos_2 = 0;
+        throttle.throttle_state = THROTTLE_OUT_OF_RANGE;
+        throttle_debug.throttle_out_of_range = true;
+        implausibility = true;
+    } else {
+        // Will be reset if Ready to Drive is not on
+        throttle.throttle_state = THROTTLE_RUN;
+        throttle_debug.throttle_out_of_range = false;
+    }
+    return implausibility;
```

Comment:
With your out of range debug message, if the first position is out of range, but the second isn't, you're overriding the `throttle_debug.throttle_out_of_range` value, meaning that the CAN message won't say there's an out of range error even if there is. I suggest splitting up the out of range signal into two separate values, one for each pot, in the CAN message.

Reply:
Same for `throttle.throttle_state`